dace/transformation/dataflow/buffer_tiling.py

```diff
@@ -0,0 +1,134 @@
+        map1_exit = graph.nodes()[candidate[BufferTiling._map1_exit]]
+        map2_entry = graph.nodes()[candidate[BufferTiling._map2_entry]]
+
+        # Check that locally provided buffer has volume 1.
+        for edge in graph.in_edges(map1_exit):
+            if edge.data.volume != 1:
+                return False
+
+        # Check that all locally provided buffers have the same subset.
+        subsets = set(edge.data.subset for edge in graph.in_edges(map1_exit))
+        if len(subsets) != 1:
+            return False
+
+        # Check existence of buffers
+        buffers = graph.all_nodes_between(map1_exit, map2_entry)
+        if buffers is None:
+            return False
+
+        for buf in buffers:
+            # Check that buffers are AccessNodes.
+            if not isinstance(buf, nodes.AccessNode):
+                return False
+
+            # Check that buffers are transient.
+            if not sdfg.arrays[buf.data].transient:
+                return False
+            
+            # Check that buffers have exactly 1 input and 1 output edge.
+            if graph.in_degree(buf) != 1:
+                return False
+            if graph.out_degree(buf) != 1:
+                return False
+            
+            # Check that buffers are next to the maps.
+            if graph.in_edges(buf)[0].src != map1_exit:
+                return False
+            if graph.out_edges(buf)[0].dst != map2_entry:
+                return False
+
+            # Check that the data consumed is provided.
+            provided = graph.in_edges(buf)[0].data.subset
+            consumed = graph.out_edges(buf)[0].data.subset
+            if not provided.covers(consumed):
+                return False
+
+            # Check that buffers occure only once in this state.
+            num_occurrences = len([
+                    n for n in graph.nodes()
+                    if isinstance(n, nodes.AccessNode) and n.data == buf
+                ])
+            if num_occurrences > 1:
+                return False
+        return True
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        map1_exit = graph.nodes()[candidate[BufferTiling._map1_exit]]
+        map2_entry = graph.nodes()[candidate[BufferTiling._map2_entry]]
+
+        return " -> ".join(entry.map.label + ": " + str(entry.map.params)
+                           for entry in [map1_exit, map2_entry])
+
+    def apply(self, sdfg):
+        graph = sdfg.nodes()[self.state_id]
+        map1_exit = graph.nodes()[self.subgraph[self._map1_exit]]
+        map1_entry = graph.entry_node(map1_exit)
+        map2_entry = graph.nodes()[self.subgraph[self._map2_entry]]
+        buffers = graph.all_nodes_between(map1_exit, map2_entry)
+        # Situation:
+        # -> map1_entry -> ... -> map1_exit -> buffers -> map2_entry -> ...
+
+        lower_extents = tuple(b[0]-a[0] for a,b in zip(map1_entry.range.ranges, map2_entry.range.ranges))
```

Comment:
there's a function called `min_element` for subsets, which should be helpful here.